### ISSUE TITLE
Explore: datasource selection happens when selector was closed

### ISSUE THREAD

**Issue by davkal** [grafana/grafana#16154]
**What happened**:
Datasource selected when focussing on query field in Explore

**What you expected to happen**:
Focus on query field.

**How to reproduce it (as minimally and precisely as possible)**:

1) Run a query in explore
2) open datasource selector
3) click browser back, notice the selector being closed.
4) click in query field

Result: a different datasource is being selected from the dropdown that should no longer be visible. 

**Anything else we need to know?**:

Needs enough datasources configured so that the list of datasources overlaps with the query field.

**Comment by hugohaggmark** [comment]
Do we really need to open the datasource select/dropdown menu on focus?
https://github.com/grafana/grafana/blob/99e4bbc23eb9784250f905d44e12a0a67eda947b/public/app/core/components/Select/DataSourcePicker.tsx#L61

**Comment by davkal** [comment]
> Do we really need to open the datasource select/dropdown menu on focus?

@torkelo added this. IMO we dont need it.
For reference, classic HTML dropdowns highlight on focus, and then you need to press Space to open them.

**Comment by torkelo** [comment]
> For reference, classic HTML dropdowns highlight on focus, and then you need to press Space to open them.

Agree, we do not need the open on focus, might have set to true without really thinking.
